src/main/kotlin/com/sourcegraph/cody/agent/EnumTypeAdapterFactory.kt

```diff
@@ -0,0 +1,57 @@
+package com.sourcegraph.cody.agent
+
+import com.google.gson.*
+import com.google.gson.annotations.SerializedName
+import com.google.gson.reflect.TypeToken
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonWriter
+import java.io.IOException
+import java.lang.reflect.Field
+
+class EnumTypeAdapterFactory : TypeAdapterFactory {
```

Comment:
Does this fix the issue in lsp4j where it writes numbers for string enums?

Reply:
Yes

Reply:
We could probably just fix this in codegen too by generating serializers for all the enums

Reply:
I'd prefer not to do that because we generate a lot of enums. I have a different workaround in lsp4j. If this adapter does the job then it's fine